### PR TITLE
Modify Jenkinsfile to perform a signed rpm release

### PR DESCRIPTION
**Pull request by cmcginty**
- Extract the PKG-INFO file from docker image
- Add release-signed make target for running Box release script

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @cmcginty has signed the CLA. Thanks for the pull request!

File: Jenkinsfile [hunk 1]
```diff
@@ -1,17 +1,21 @@
 pipeline {
-    agent { label 'rosie' }
+    agent none
```
Comment (ethomas2) [inline Jenkinsfile]: I see agent is being overridden below, but can you put a default agent here anyway? So someone doesn't add a new stage and accidentally run on some random node

File: Jenkinsfile [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 
     stages {
         stage('Build') {
+            agent { label 'rosie' }
             steps {
-                sh 'make docker-rpm'
+                sh 'make clean docker-rpm'
+                stash name: 'rpm', includes: 'dist/*.rpm'
```
Comment (ethomas2) [inline Jenkinsfile]: What do stash and unstash do? If they're relatively uncommon commands then add a comment please. If they're common and I'm just not familiar with jenkinsfiles then just leave it alone

File: Makefile [hunk 3]
```diff
@@ -175,18 +175,22 @@ rpm: $(CR_BIN) $(PY_PKG_INFO)
...
-	mkdir -p $(DIST_DIR)
+	mkdir -p $(DIST_DIR) $(PY_EGG_INFO_DIR)
 	@# Docker cp does not support globing, so the path to the RPM file must be
 	@# detected with a query. The order of the commands are important and they
 	@# must all be run in the same "shell" for the variables to be available to
 	@# the final "docker cp" command.
 	DOCKER_RPM_PATH=$$(docker run $(DOCKER_TAG) sh -c "ls /root/$(DIST_DIR)/*.rpm") && \
 	CONTAINER_ID=$$(docker ps -alq) && \
 	docker cp $$CONTAINER_ID:$$DOCKER_RPM_PATH $(DIST_DIR) && \
+	docker cp $$CONTAINER_ID:/root/$(PY_PKG_INFO) $(PY_EGG_INFO_DIR) && \
```
Comment (ethomas2) [inline Makefile]: Are we copying python eggs out of the container? If so why?

File: Makefile [hunk 4]
```diff
@@ -208,6 +212,16 @@ release: $(PY_PKG_INFO) $$($$RPM_PATH)
 docker-release: docker-rpm
 	docker run --rm -e ARTIFACTORY=$(ARTIFACTORY) $(DOCKER_TAG) /usr/bin/make release
 
+# See "release" target. The major difference is that this target uses the Box custom
+# rpm-to-artifactory tooling that is made available for signing and uploading rpms.
+.PHONY: release-signed
+release-signed: $(PY_PKG_INFO) $$($$RPM_PATH)
```
Comment (ethomas2) [inline Makefile]: Can we get rid of release and docker-release now?

Comment (cmcginty) [reply Makefile]: Also it is more useful for users outside of Box.

**Pull request merged**